deeplake/core/vectorstore/vector_search/dataset/dataset.py

```diff
@@ -63,116 +106,78 @@ def load_dataset(dataset_path, token, creds, logger, read_only, **kwargs):
     return dataset
 
 
-def create_tensors_if_needed(dataset, logger):
+def check_tensors(dataset):
     tensors = dataset.tensors
 
-    if "text" not in tensors:
-        warn_and_create_missing_tensor(
-            logger=logger,
-            dataset=dataset,
-            tensor_name="text",
-            htype="text",
-            create_id_tensor=False,
-            create_sample_info_tensor=False,
-            create_shape_tensor=False,
-            chunk_compression="lz4",
-        )
+    embedding_htype_exist = False
+    ids_exist = False
 
-    if "metadata" not in tensors:
-        warn_and_create_missing_tensor(
-            logger=logger,
-            dataset=dataset,
-            tensor_name="metadata",
-            htype="json",
-            create_id_tensor=False,
-            create_sample_info_tensor=False,
-            create_shape_tensor=False,
-            chunk_compression="lz4",
-        )
+    for tensor in tensors:
+        if tensor in ("id", "ids"):
+            ids_exist = True
 
-    if "embedding" not in tensors:
-        warn_and_create_missing_tensor(
-            logger=logger,
-            dataset=dataset,
-            tensor_name="embedding",
-            htype="embedding",
-            dtype=np.float32,
-            create_id_tensor=False,
-            create_sample_info_tensor=False,
-            max_chunk_size=64 * MB,
-            create_shape_tensor=True,
-        )
+        if tensor in ("embedding", "embeddings"):
+            embedding_htype_exist = True
 
-    if "ids" not in tensors:
-        warn_and_create_missing_tensor(
-            logger=logger,
-            dataset=dataset,
-            tensor_name="ids",
-            htype="text",
-            create_id_tensor=False,
-            create_sample_info_tensor=False,
-            create_shape_tensor=False,
-            chunk_compression="lz4",
-        )
+        htype = dataset[tensor].htype
+        if htype == "embedding":
```

Comment:
If a tensor with htype embedding exists but it is named something else, do we pick that up? Also consider edge case where tensor name is id and htype is embedding. Also case where name is embedding but htype is not supported (text, json etc)

Reply:
yup. we pick that. will add edge cases